lib/doorkeeper/openid_connect/helpers/controller.rb

```diff
@@ -38,17 +38,20 @@ def handle_oidc_error!(exception)
           # FIXME: workaround for Rails 5, see https://github.com/rails/rails/issues/25106
           @_response_body = nil
 
+          should_respond_on_fragment = Doorkeeper::OpenidConnect::ResponseMode.new(pre_auth.response_type).fragment?
```

Comment:
Could we just call `pre_auth.response_on_fragment?` here, and make that method public?

Reply:
Sure! I didn't want to mess about with the api given my low familiarity with the code base. I'll fix that.